addons/binding/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/handler/AirVisualNodeHandler.java

```diff
@@ -0,0 +1,234 @@
+            nodeData = currentNodeData;
+            // Update all channels from the updated Node data
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId(), false);
+            }
+        }
+    }
+
+    private String getNodeJsonData() throws IOException {
+        String url = "smb://" + nodeAddress +"/" + nodeShareName + "/" + NODE_JSON_FILE;
+        NtlmPasswordAuthentication auth = new NtlmPasswordAuthentication(null, nodeUsername, nodePassword);
+        try (SmbFileInputStream in = new SmbFileInputStream(new SmbFile(url, auth))) {
+            return IOUtils.toString(in, StandardCharsets.UTF_8.name());
+        }
+    }
+
+    private void updateChannel(String channelId, boolean force) {
+        if (nodeData != null && (force || isLinked(channelId))) {
+            State state = getChannelState(channelId, nodeData);
+            logger.debug("Update channel {} with state {}", channelId, state);
+            updateState(channelId, state);
+        }
+    }
+
+    private State getChannelState(String channelId, NodeData nodeData) {
+        State state = UnDefType.UNDEF;
+
+        // Handle system channel IDs separately, because 'switch/case' expressions must be constant expressions
+        if (CHANNEL_BATTERY_LEVEL.equals(channelId)) {
+            state = new DecimalType(BigDecimal.valueOf(nodeData.getStatus().getBattery()).longValue());
+        } else if (CHANNEL_WIFI_STRENGTH.equals(channelId)) {
+            state = new DecimalType(BigDecimal.valueOf(Math.max(0, nodeData.getStatus().getWifiStrength()-1)).longValue());
+        } else {
+            // Handle binding-specific channel IDs
+            switch (channelId) {
+                case CHANNEL_CO2:
+                    state = new QuantityType<>(nodeData.getMeasurements().getCo2Ppm(), PARTS_PER_MILLION);
+                    break;
+                case CHANNEL_HUMIDITY:
+                    state = new QuantityType<>(nodeData.getMeasurements().getHumidityRH(), PERCENT);
+                    break;
+                case CHANNEL_AQI_US:
+                    state = new QuantityType<>(nodeData.getMeasurements().getPm25AQIUS(), ONE);
+                    break;
+                case CHANNEL_PM_25:
+                    // PM2.5 is in ug/m3
+                    state = new QuantityType<>(nodeData.getMeasurements().getPm25Ugm3(), MICRO(GRAM).divide(CUBIC_METRE));
+                    break;
+                case CHANNEL_TEMP_CELSIUS:
+                    state = new QuantityType<>(nodeData.getMeasurements().getTemperatureC(), CELSIUS);
+                    break;
+                case CHANNEL_TIMESTAMP:
+                    // It seem the Node timestamp is Unix timestamp converted from UTC time plus timezone offset.
+                    // Not sure about DST though, but it's best guess at now
+                    Calendar cal = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
```

Comment:
We have more modern options since Java 8, I would prefer if you use them instead of the calendar